### PR TITLE
Parse package version installed as editable

### PR DESCRIPTION
**Pull request by sbrunner**
See tests or https://geomapfish-demo-2-5.camptocamp.com/c2c/versions.json for more information

### CODE REVIEW

**Comment by gberaudo** [comment]
@sbrunner

-  why do you need this?
- is the format stable? (I checked https://pip.pypa.io/en/stable/reference/pip_freeze/ then https://pip.pypa.io/en/stable/reference/pip_install/#requirements-file-format and I see nothing about this editable syntax)

**Comment by sbrunner** [comment]
Because if you see the provided link the actual versions are broken, and it's the only way that I see to get the version of package installed from source (that's effectively not documented...).

File: c2cwsgiutils/scripts/genversion.py [hunk 1]
```diff
@@ -4,9 +4,21 @@
...
 import sys
-from typing import Dict
+from typing import Dict, Optional, Tuple, cast
 
-VERSION_RE = re.compile(r"^(.*)==(.*)$")
+SRC_VERSION_RE = re.compile(r"^.*\(([^=]*)===?([^=]*)\)$")
+VERSION_RE = re.compile(r"^([^=]*)==([^=]*)$")
+
+
+def _get_package_version(comp: str) -> Tuple[Optional[str], Optional[str]]:
```
Comment (gberaudo) [inline c2cwsgiutils/scripts/genversion.py:13]: Document parser
```suggestion
def _get_package_version(comp: str) -> Tuple[Optional[str], Optional[str]]:
"""
Parse plain and editable versions. See test_genversion.py for examples.
"""
```

Comment (sbrunner) [reply c2cwsgiutils/scripts/genversion.py:13]: Done, thanks :-)

**Pull request merged**